src/main/java/com/mathworks/ci/MatlabItemListener.java

```diff
@@ -0,0 +1,91 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2020 The MathWorks, Inc.
+ *
+ * Item listener class to provide functionality to check UI element states for a
+ * Multi-configuration project.
+ *
+ */
+
+import hudson.Extension;
+import hudson.matrix.MatrixConfiguration;
+import hudson.matrix.MatrixProject;
+import hudson.model.Item;
+import hudson.model.TopLevelItem;
+import hudson.model.listeners.ItemListener;
+import jenkins.model.Jenkins;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Extension
+public final class MatlabItemListener extends ItemListener {
+    private static final Map<String, Boolean> prjCheckMATLABAxis = new HashMap<>();
+    private static final Map<String, Boolean> prjCheckMATLABBuildWrapper = new HashMap<>();
+
+    @Override
+    public void onLoaded(){
+        checkItems(Jenkins.get().getItems());
+    }
+
+    @Override
+    public void onUpdated(Item item) {
+        if(!(item instanceof MatrixProject)){
+            return;
+        }
+        checkSingleItem(item);
+    }
+
+
+    private void checkItems(List<TopLevelItem> items) {
+        for(TopLevelItem item : items){
+            if(item instanceof MatrixProject){
+                check((MatrixProject) item);
+            }
+        }
+    }
+
+    private void checkSingleItem(Item item){
+        check((MatrixProject) item);
+    }
+
+    private void check(MatrixProject _prj) {
+        checkForAxis(_prj);
+        checkForBuildWrapper(_prj);
+    }
+
+    private void checkForAxis(MatrixProject _prj) {
+        boolean checkForAxis = false;
+        Collection<MatrixConfiguration> conf = _prj.getActiveConfigurations();
+        for(MatrixConfiguration _conf : conf){
+            String a = _conf.getCombination().get(Message.getValue("Axis.matlab.key"));
```

Comment:
String 'a' is not intuitive about what it is used for.

Reply:
Done.